tests/test_pump_scheduler.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import absolute_import
+import unittest
+import datetime
+
+import mock
+
+from greenpithumb import pump_scheduler
+
+
+class PumpSchedulerTest(unittest.TestCase):
+
+    def setUp(self):
+        self.mock_clock = mock.Mock()
+
+    def test_wake_hour_less_than_sleep_hour(self):
+        """Test rollover from one day to the next.
+        
+        If the sleep period spans two days, the pump should not run between 
+        midnight and the wake hour.
+        """
+        self.mock_clock.now.return_value = datetime.datetime(2016, 5, 24, 0)
+        sleep_windows = ((22, 8),)
+        mock_pump_scheduler = pump_scheduler.PumpScheduler(self.mock_clock,
+                                                           sleep_windows)
+        self.assertFalse(mock_pump_scheduler.is_running_pump_allowed())
+
+    def test_wake_hour_greater_than_sleep_hour(self):
+        self.mock_clock.now.return_value = datetime.datetime(2016, 5, 24, 4)
+        sleep_windows = ((2, 8),)
+        mock_pump_scheduler = pump_scheduler.PumpScheduler(self.mock_clock,
+                                                           sleep_windows)
+        self.assertFalse(mock_pump_scheduler.is_running_pump_allowed())
+
+    def test_multiple_sleep_windows(self):
+        """Pump should not run if current hour is in any sleep window."""
+        self.mock_clock.now.return_value = datetime.datetime(2016, 5, 24, 4)
+        sleep_windows = ((16, 20), (2, 8))
+        mock_pump_scheduler = pump_scheduler.PumpScheduler(self.mock_clock,
+                                                           sleep_windows)
+        self.assertFalse(mock_pump_scheduler.is_running_pump_allowed())
+
+    def test_running_pump_allowed_if_not_in_sleep_window(self):
+        self.mock_clock.now.return_value = datetime.datetime(2016, 5, 24, 10)
+        sleep_windows = ((2, 8),)
+        mock_pump_scheduler = pump_scheduler.PumpScheduler(self.mock_clock,
+                                                           sleep_windows)
+        self.assertTrue(mock_pump_scheduler.is_running_pump_allowed())
```

Comment:
Suggest adding a test case for when the current hour is the sleep hour and and one where the current hour is the wake hour.